### PR TITLE
Initial release + tests

### PR DESCRIPTION
**Pull request by kosciCZ**
This PR would add the ability for the bot to do release from the scratch - meaning the package wasn't previously released on PyPi or Github. This required mostly small changes to existing functions and logic of the bot. 

List of more significant changes:
- separated github and pypi releasing - github release doesn't depend on pypi version number anymore
- `Github.latest_release()` now always returns either version number or None - ignores prereleases and searches for latest released version 
- fixed wrong http method in `Github.branch_exists()`
- changelog is not git added if it wouldn't change

I decided to implement this feature partly because I felt it might benefit users (deployment to a new repo becomes easier) and partly because I wanted to finally have some end-to-end tests implemented the way I proposed [here](https://github.com/user-cont/release-bot/issues/65#issuecomment-412448582).
The tests this PR is adding are not truly end-to-end, because the bot runs in a while loop, but I tried to simulate the bot workflow as much as possible.   

This would solve #65.

**Edit:** I have fixed numerous code style issues, but some remained, because they can't be fixed in context of pytest (class variables defined outside` __init__`), unused arguments (pytest fixtures), some 'unable to import' warnings and 'Instance of 'PurePath' has no 'read_text' member' bug from old version of pytest.

### CODE REVIEW

File: tests/test_bot.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+import pytest
+import yaml
+from flexmock import flexmock
+
+from release_bot.configuration import configuration
+from release_bot.releasebot import ReleaseBot
+from github_utils import GithubUtils, RELEASE_CONF
+
+
+@pytest.mark.skipif(not GithubUtils.github_api_status(), reason="Github api is down")
```
Comment (jpopelka) [inline tests/test_bot.py]: nice

File: tests/test_bot.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    @pytest.fixture()
+    def open_pr(self):
+        """Opens two release issues in a repository"""
+        conf = yaml.safe_load(RELEASE_CONF) or {}
+        self.release_bot.new_release.update(conf)
+        self.open_issue()
+        self.release_bot.find_open_release_issues()
+        self.release_bot.make_release_pull_request()
+        pr_number = self.release_bot.github.pr_exists("0.0.1 release")
+        assert pr_number
```
Comment (jpopelka) [inline tests/test_bot.py]: merge these 2 lines

File: tests/src/example_setup.py.txt [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+setuptools.setup(
+    name="{}",
+    version=release_bot_test.__version__,
+    author="User Cont",
+    author_email="user-cont@redhat.com",
+    description="A small example package used fo release-bot testing",
+    long_description="Test",
+    long_description_content_type="text/markdown",
+    url="https://github.com/{}/{}",
```
Comment (jpopelka) [inline tests/src/example_setup.py.txt]: Would it be possible to use
```python
url="https://github.com/{user}/{repo}",
```
in order to use kwargs later in `format()` ?

File: tests/test_bot.py [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        """Setups environment for releasing on Github"""
+        self.open_pr()
+        assert self.release_bot.find_newest_release_pull_request()
+        self.release_bot.make_new_github_release()
+        assert self.release_bot.github.latest_release() == "0.0.1"
+
+    @pytest.fixture()
+    def mock_upload(self):
+        """Mocks upload to PyPi"""
+        flexmock(self.release_bot.pypi, upload=True)
```
Comment (jpopelka) [inline tests/test_bot.py]: Am I correct that this is a [shorthand](https://flexmock.readthedocs.io/en/latest/user-guide/#shorthand) to
```python
flexmock(self.release_bot.pypi).should_receive('upload').and_return('True')
```
?

File: .travis.yml [hunk 5]
```diff
@@ -1,10 +1,12 @@
...
+  - secure: kHLBmNa39znq9d+Egl5BaxQFaEhC/igKTzK2XpGDwovkDlGpi9tc5qq21vLVXilx93MFQdWt2RwbU7SPVc6WtMGUdIGl8pq3ebiaBj6TPeoY/f5zTAtTV2uDrBid1jCeaolpwxx21SoYda+0dBDeBTTo8bWEiu3ei1gB9mLiR3sJsmMjqoaKsMDGLnh+iLlDZj67lWNd6itee4E7sm+38Uac1oRa4dM+h6jS5lJ+ceO+sezNYR8qVz81XLOlE6h0O8+Um1/QqTro7J31DRewbxv1I0R/ut4by9ANNts75JGuEbPF2CaxaK5rTX/kORTCq0ngEOhpY3Lphe1JNfkqdPmnIBwooKcG+HQkjXRG0MxOVT/SHycvt25FyGP7KkLpxJPFx4OMg59k/k0I487EVwzxXIVuIW0aEFsneTqdhTRMRc4Vq5ADoWAd9DQHP78tGUwppjZ3P3vEVv3ArPPmpNxLfP7igBa0CSULm1dEr3I8A6i5YT1gtPiSYfcQcd/my6RGvyWDzAi3vhAEJfA9DbQHreMEHRCLnYwgOVelRuyti8tzSXi3RaiW9KE6JI599oIgNhxmA2ezhIqNH3pFqMehPtpvN0hsXv1Ek5XLcD3CRokdXfE4DsJiV9eDem0sEnsh5o9FbnR26WvMT7qYJxuVigBZDhkflrMrgbpdlGk=
```
[the remark this answers is not in the document]
Comment (kosciCZ) [reply .travis.yml]: No, this is a separate token I've created on our bot account for CI purposes.

File: tests/test_bot.py (hunk 4 above)
Comment (kosciCZ) [reply tests/test_bot.py]: Yes it is.

**Pull request merged**